core/src/DevStep.cpp

```diff
@@ -51,7 +51,7 @@ void DevStep::iterate(const TimestepTime& tst)
     OutputSpec os; // The default OutputSpec is all fields, but only cell average values
     ModelState overallState = pData->getStateRecursive(os);
     overallState.merge(ConfiguredModule::getAllModuleConfigurations());
-    Module::getImplementation<IDiagnosticOutput>().outputState(*mData);
+    Module::getImplementation<IDiagnosticOutput>().outputState(*mData, tst.step);
```

Comment:
Suggest reading the configuration at `ModelConfig::TIMESTEP_KEY` rather than introducing a new argument into `outputState()`.

Reply:
Then I should use ``Configured::getConfiguration`` and parse the string, right? It seems like the wrong place to do that. I'd rather add the time step to the metadata store. Seems like it should be there anyway.